Src/CSharpier/SyntaxPrinter/RightHandSide.cs

```diff
@@ -8,40 +9,59 @@ namespace CSharpier.SyntaxPrinter
+        public static Doc Print(Doc leftDoc, Doc operatorDoc, ExpressionSyntax rightNode)
         {
+            var layout = DetermineLayout(rightNode);
+
             var groupId = Guid.NewGuid().ToString();
-            return node switch
+
+            return layout switch
             {
-                InitializerExpressionSyntax initializerExpressionSyntax
-                  => InitializerExpression.Print(initializerExpressionSyntax),
-                InvocationExpressionSyntax
-                or ParenthesizedLambdaExpressionSyntax
-                or ObjectCreationExpressionSyntax
-                or ElementAccessExpressionSyntax
-                or ArrayCreationExpressionSyntax
-                or ImplicitArrayCreationExpressionSyntax
+                Layout.BasicConcatWithoutLine
+                  => Doc.Concat(leftDoc, operatorDoc, Node.Print(rightNode)),
+                Layout.BreakAfterOperator
+                  => Doc.Group(
+                      Doc.Group(leftDoc),
+                      operatorDoc,
+                      Doc.Group(Doc.Indent(Doc.Line, Node.Print(rightNode)))
+                  ),
+                Layout.Fluid
                   => Doc.Group(
+                      Doc.Group(leftDoc),
+                      operatorDoc,
                       Doc.GroupWithId(groupId, Doc.Indent(Doc.Line)),
-                      Doc.IndentIfBreak(Doc.Group(Node.Print(node)), groupId)
+                      Doc.IndentIfBreak(Node.Print(rightNode), groupId)
                   ),
-                AnonymousObjectCreationExpressionSyntax
-                or AnonymousMethodExpressionSyntax
-                or ConditionalExpressionSyntax
-                or SwitchExpressionSyntax
-                or LambdaExpressionSyntax
-                or AwaitExpressionSyntax
-                or WithExpressionSyntax
-                  => Doc.Group(Doc.Group(Doc.Indent(Doc.Line)), Node.Print(node)),
+                _ => throw new Exception("The layout type of " + layout + " was not handled.")
+            };
+        }
+
+        private static Layout DetermineLayout(ExpressionSyntax rightNode)
+        {
+            return rightNode switch
+            {
+                InitializerExpressionSyntax => Layout.BasicConcatWithoutLine,
                 BinaryExpressionSyntax
+                or ImplicitObjectCreationExpressionSyntax
                 or InterpolatedStringExpressionSyntax
                 or IsPatternExpressionSyntax
                 or LiteralExpressionSyntax
                 or QueryExpressionSyntax
                 or StackAllocArrayCreationExpressionSyntax
-                  => Doc.Indent(Doc.Group(Doc.Line, Node.Print(node))),
-                _ => Doc.Group(Doc.Indent(Doc.Line), Doc.Indent(Node.Print(node)))
+                or MemberAccessExpressionSyntax
+                or ConditionalExpressionSyntax { Condition: BinaryExpressionSyntax
+                or ParenthesizedExpressionSyntax }
```

Comment:
Does this have its own issue? It shouldn't be aligned with the rest, since it's nested in a sub object

Reply:
I added it on to #154